cmd/tink-worker/internal/registry.go

```diff
@@ -64,8 +74,17 @@ func (r *RegistryConnDetails) pullImage(ctx context.Context, cli *client.Client,
 		return errors.Wrap(err, "DOCKER PULL")
 	}
 	defer out.Close()
-	if _, err := io.Copy(os.Stdout, out); err != nil {
-		return err
+	fd := json.NewDecoder(out)
+	var status *ImagePullStatus
+	for {
+		if err := fd.Decode(&status); err != nil {
+			if err == io.EOF {
+				break
+			}
+		}
+		if status.Error != "" {
```

Comment:
If error occurs but it's not EOF, deferencing `status` for `Error` field will panic.

Reply:
okay, I have added a return statement if `Decode` fails.